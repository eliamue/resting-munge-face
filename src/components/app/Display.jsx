import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ onChange, onSubmit }) => {
  return (
    <>
      <form onSubmit={onSubmit}>
        <label>GET</label>
        <input 
          type="radio"
          id="get"
          onChange={onChange}
        />

        <label>PUT</label>
        <input 
          type="radio"
          id="put"
          onChange={onChange}
        />

        <label>POST</label>
        <input 
          type="radio"
          id="post"
          onChange={onChange}
        />

        <label>DELETE</label>
        <input 
          type="radio"
          id="delete"
          onChange={onChange}
        />
        <button aria-label="submit-request">Submit</button>
      </form>
    </>
  );
};

Display.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default Display;
