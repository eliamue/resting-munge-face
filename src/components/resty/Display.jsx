import React from 'react';
import PropTypes from 'prop-types';
import ReactJson from 'react-json-view';

const Display = (res) => {
  return (
    <div>
      <ReactJson src={res} />
    </div>
  );
};

Display.propTypes = {
  res: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default Display;
