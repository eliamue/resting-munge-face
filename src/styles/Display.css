.Display {
    display: flex;
    background-color: white;
    border: 1px rgb(170, 169, 169) solid;
    padding: 5px;
    margin: 5px;
    text-align: flex-start;
    width: 80%;
    margin-left: 150px;
    height: 60vmin;
}
