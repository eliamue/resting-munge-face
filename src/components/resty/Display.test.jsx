import React from 'react';
import { render, screen } from '@testing-library/react';
import Display from './Display';

describe('Display', () => {
  it('displays test body', () => {
    const fetch = { Henlo: 'Go fetch!' };
    render(<Display display={fetch} />);

    const display = screen.getByText('"Go fetch!"');
    expect(display).toMatchSnapshot();
  });

});
