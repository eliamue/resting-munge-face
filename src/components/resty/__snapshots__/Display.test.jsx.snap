// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display displays test body 1`] = `
<span
  class="string-value"
  style="cursor: default;"
>
  "
  Go fetch!
  "
</span>
`;
